lib/private/Files/ObjectStore/ObjectStoreStorage.php

```diff
@@ -596,6 +596,13 @@ public function copyFromStorage(
 
 	public function moveFromStorage(IStorage $sourceStorage, $sourceInternalPath, $targetInternalPath, ?ICacheEntry $sourceCacheEntry = null): bool {
 		$sourceCache = $sourceStorage->getCache();
+		if ($sourceStorage->instanceOfStorage(ObjectStoreStorage::class)) {
+			if ($sourceStorage->getObjectStore()->getStorageId() === $this->getObjectStore()->getStorageId()) {
```

Comment:
```suggestion
		if ($sourceStorage->getId() === $this->getId()) {
```
Judging from the docs for IStorage::getId() this should be enough, although I don't know if there is a guarantee two different storages will absolutely never have the same ID, so your approach would be a lot safer then. @icewind1991 should know this in more detail.

Reply:
I had the same idea, too. In the groupfolders scenario, source and target storage are represented by different objects, both having a different type: One is of the type `HomeObjectStoreStorage`, the other one of type `ObjectStoreStorage`. The identifiers returned by `getId()` are different. However, when you have an object store as primary storage, both storage objects use the same underlying bucket. In order to cover this scenario, I think using the bucket identifier returned by `IObjectStore#getStorageId()` is a safe method for checking bucket equality.

Reply:
Ok makes sense! Please apply the other suggestion I gave below then.